### ISSUE TITLE
bug: constraint attr.consul.version semver >= 1.7.0 is missing after nomad 1.7 update

### ISSUE THREAD

**Issue by Kamilcuk** [hashicorp/nomad#22190]
### Nomad version
v1.7.5

### Operating system and Environment details
Fedora 29

### Issue
When starting a job with service, it was auto-added a constraint that consul has to exist on the machine with Nomad. Now that constraint is not added.

### Reproduction steps
1. Run a job with consul service
2. Note the constraint is missing

#### Expected Result
There should be the constraint.

#### Actual Result
The constraint is no longer there. Job file has not changed.

```
+/- Job: "tabix"
+/- Task Group: "tabix" (1 create/destroy update)
  -   Constraint {
      - LTarget: "${attr.consul.version}"
      - Operand: "semver"
      - RTarget: ">= 1.7.0"
      }
  +   Consul {
      + Cluster:   "default"
        Namespace: ""
        Partition: ""
      }
  +/- Task: "tabix-db" (forces in-place update)
    +/- Service {
        Address:           ""
        AddressMode:       "auto"
      + Cluster:           "default"
        EnableTagOverride: "false"
        Name:              "tabix"
        Namespace:         "default"
        OnUpdate:          "require_healthy"
        PortLabel:         "www"
        Provider:          "consul"
        TaskName:          "tabix-db"
        Tags {
          Tags: "urlprefix-tabix.service.consul.avt.striketechnologies.com/"
        }
        }
```

### Job file (if appropriate)

```
job "tabix" {
  group "tabix" {
    network {
      port "www" { to = 80 }
    }
    task "tabix" {
      driver = "docker"
      config {
        image = "busybox"
        ports = ["www"]
      }
      service {
        name = "tabix"
        tags = ["urlprefix-tabix.service.consul.avt.striketechnologies.com/"]
        port = "www"
        check {
          type     = "tcp"
          interval = "5s"
          timeout  = "1s"
        }
      }
    }
  }
}
```

Hi. Please kindly let me know if this is intended, and I should now add the constraint manually to jobs that I want to run with consul, or this is unintended and something is maybe wrong with my configuration. Thanks.

**Comment by pkazmierczak** [comment]
Hey @Kamilcuk, I can reproduce and indeed confirm that this is a regression, possibly something we introduced during recent Consul integration revamp. I'll have a look and report back!

**Comment by Kamilcuk** [comment]
Supcio, dzięki!

**Comment by pkazmierczak** [comment]
The fix will land in 1.8 which will be released soon. Thanks again for reporting this!
